### PR TITLE
feat(upgrade): add upgradeTask CR definition

### PR DESCRIPTION
**Pull request by shubham14bajpai**
Signed-off-by: shubham <shubham.bajpai@mayadata.io>

This PR adds the CR definition for `UpgradeTask`. UpgradeTask is a CR used to upgrade a resource using the upgrade job and get the updates of each step through the CR. The upgrade job can list UpgradeTask CR via labels and perform multiple upgrades in a single job.

### CODE REVIEW

File: pkg/apis/openebs.io/v1alpha1/upgradetask_types.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+	PreUpgrade UpgradeStep = "PRE_UPGRADE"
+	// TargetUpgrade is the step to upgrade Target depoyment of resource
+	TargetUpgrade UpgradeStep = "TARGET_UPGRADE"
+	// ReplicaUpgrade is the step to upgrade replica deployment of resource
+	ReplicaUpgrade UpgradeStep = "REPLICA_UPGRADE"
+	// Verify is the step to verify the upgrade
+	Verify UpgradeStep = "VERIFY"
+	// Rollback is the step to rollback to previous version if upgrade fails
+	Rollback UpgradeStep = "ROLLBACK"
+	// PoolInstanceUpgrade is the step to verify resource before upgrade
```
Comment (sagarkrsd) [inline pkg/apis/openebs.io/v1alpha1/upgradetask_types.go]: Can this comment be more detailed or explanatory, I didn't get a hold of it.

Comment (sagarkrsd) [reply pkg/apis/openebs.io/v1alpha1/upgradetask_types.go]: Also, Is this is going to be the only new step that will be included for CSPC and CSPI upgrade status?

Comment (shubham14bajpai) [reply pkg/apis/openebs.io/v1alpha1/upgradetask_types.go]: This step is present in old upgrade also. As the pool does not have any target or replica terms attached to it we have this step for each CSP. The same can be used to represent CSPI upgrade.
https://github.com/openebs/maya/blob/e69d3976173625c05b09da74fcd86e491a0bee2a/pkg/upgrade/upgrader/csp_upgrade.go#L307

**Review by sagarkrsd — APPROVED** [review]

**Review by prateekpandey14 — APPROVED** [review]
lgtm

**Review by kmova — APPROVED** [review]

**Pull request merged**